### DISCUSSION TITLE
LSB shared object for arm 32-bit

### DISCUSSION THREAD

**Discussion by hemanthofficial3009** [vectordotdev/vector#11958] [General]
Hello team,
 When i tried to download the vector binary for arm 32-bit architecture (version 0.18.1), im running in to the following issue: "_vector: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 3.2.0, BuildID[sha1]=414d8e9563cf09861560c79fa80a033d95741805, with debug_info, not stripped_".

Can you please help us understand what was the issue in terms of LSB Shared object and not the executable?

**Comment by jszwedko** [comment]
Hi @hemanthofficial3009 !

Are you having trouble running the binary? https://unix.stackexchange.com/a/512509/108878 makes it seem like it'll be reported as a shared object if it is able to have its addresses randomized, even if it is an executable.

**Reply by jszwedko** [reply]
See also: https://www.reddit.com/r/rust/comments/d97j7w/cargo_generates_shared_object_instead_of/

**Reply by bmallya** [reply]
@jszwedko , yes we were not able to run the arm binary. It would complain that "vector not found".

**Reply by hemanthofficial3009** [reply]
@jszwedko, 
$ uname -a
Linux XYZ 3.14.52 #2 SMP PREEMPT Fri Aug 20 23:35:01 UTC 2021 armv7l GNU/Linux
When tried to run the armv7 vector binary on the system mentioned above, it says "not found" 
![image](https://user-images.githubusercontent.com/65534235/159856133-9396dde2-6167-462a-8d97-73e004e2a113.png)

**Reply by jszwedko** [reply]
Hi @hemanthofficial3009 ! Hmm, are you sure `vector` exists at the path? Even if it isn't actually an executable, you should see a different error. That error seems to indicate `vector` doesn't exist at that path.

**Reply by hemanthofficial3009** [reply]
@jszwedko, the binary is present in the same directory.
![image](https://user-images.githubusercontent.com/65534235/159946062-7f256681-9479-418d-8c65-7fe348f5c4ab.png)

**Reply by jszwedko** [reply]
Interesting. I'm not exactly sure what might be going on here. I was able to run the armv7 binary myself for 0.18.1:

```
~ # uname -a
Linux 9a0ec7a4fd7f 4.14.232-176.381.amzn2.x86_64 #1 SMP Wed May 19 00:31:54 UTC 2021 armv7l Linux
~ # file ./vector-armv7-unknown-linux-musleabihf/bin/vector
./vector-armv7-unknown-linux-musleabihf/bin/vector: cannot open `./vector-armv7-unknown-linux-musleabihf/bin/vector' (No such file or directory)
~ # file /usr/local/bin/vector
/usr/local/bin/vector: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), statically linked, with debug_info, not stripped
~ # uname -a
Linux 9a0ec7a4fd7f 4.14.232-176.381.amzn2.x86_64 #1 SMP Wed May 19 00:31:54 UTC 2021 armv7l Linux
~ # file /usr/local/bin/vector
/usr/local/bin/vector: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), statically linked, with debug_info, not stripped
~ # /usr/local/bin/vector
2022-03-24T15:15:14.430089Z  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info"
2022-03-24T15:15:14.438563Z  INFO vector::app: Loading configs. paths=["/etc/vector/vector.toml"]
2022-03-24T15:15:14.591546Z  INFO vector::topology::running: Running healthchecks.
2022-03-24T15:15:14.596650Z  INFO vector::topology::builder: Healthcheck: Passed.
2022-03-24T15:15:14.603719Z  INFO vector::topology::running: Starting source. key=dummy_logs
2022-03-24T15:15:14.610310Z  INFO vector::topology::running: Starting transform. key=parse_logs
2022-03-24T15:15:14.614273Z  INFO vector::topology::running: Starting sink. key=print
2022-03-24T15:15:14.623519Z  INFO vector: Vector has started. debug="false" version="0.18.1" arch="arm" build_id="c4adb60 2021-11-30"
2022-03-24T15:15:14.624723Z  INFO vector::app: API is disabled, enable by setting `api.enabled` to `true` and use commands like `vector top`.
{"appname":"devankoshal","facility":"local5","hostname":"random.de","message":"#hugops to everyone who has to deal with this","msgid":"ID834","procid":2423,"severity":"debug","timestamp":"2022-03-24T15:15:14.629Z","version":2}
^C2022-03-24T15:15:15.095249Z  INFO vector: Vector has stopped.
2022-03-24T15:15:15.103442Z  INFO vector::topology::running: Shutting down... Waiting on running components. remaining_components="print, parse_logs, dummy_logs" time_remaining="59 seconds left"
^C2022-03-24T15:15:15.514375Z  INFO vector: Vector has quit.
```

Notably my `file` identifies the binary as an executable.

**Reply by jszwedko** [reply]
Ah, wait, I think I was able to reproduce it with debian.

**Reply by jszwedko** [reply]
Yeah, I was able to reproduce this on debian with our libc docker image. I'm not exactly sure what's going on, but I opened https://github.com/vectordotdev/vector/issues/11970 with some more details. The musl based builds seem to work fine if that is an option for you.

Thanks for raising this!

**Reply by jszwedko** [reply]
Also, running the armv7 docker images seems to work too.

**Reply by hemanthofficial3009** [reply]
> 

Thanks @jszwedko for looking into this real quick. Will look forward to have this resolved.

**Reply by jszwedko** [reply]
No problem! Unfortunately we don't have many leads right now. If you are familiar with ELF, we would love any thoughts 🙂
